lib/mfg.c

```diff
@@ -1041,20 +1044,26 @@ int switchtec_kmsk_set(struct switchtec_dev *dev,
 		       struct switchtec_kmsk *kmsk)
 {
 	int ret;
+	uint32_t cmd_id;
+
+	if (switchtec_is_gen5(dev))
+		cmd_id = MRPC_KMSK_ENTRY_SET_GEN5;
+	else
+		cmd_id = MRPC_KMSK_ENTRY_SET;
```

Comment:
A bit odd that the only difference between gen5 and gen4 is the command id. Normally you'd change the command ID because the interface changed some how. So I'd rather expect to see different send functions with different payloads.